bundles/CoreBundle/Migrations/Version20230616085142.php

```diff
@@ -0,0 +1,99 @@
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Pimcore\Db;
+use Pimcore\Model\Dao\AbstractDao;
+
+final class Version20230616085142 extends AbstractMigration
+{
+    private const ID_COLUMN = 'o_id';
+    private const PK_COLUMNS = [
+        self::ID_COLUMN, 'dest_id', 'type', 'fieldname', 'column', 'ownertype', 'ownername', 'position', 'index'];
+    private const UNIQUE_INDEX_NAME = 'metadata_un';
+    private const AUTO_ID = 'id';
+
+    public function getDescription(): string
+    {
+        return 'Migrate object_metadata schema to have a auto increment column';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('SET foreign_key_checks = 0');
+
+        $db = Db::get();
+
+        $metaDataTables = $db->fetchAllAssociative(
+                "SHOW FULL TABLES
+                       WHERE `Tables_in_{$db->getDatabase()}` LIKE 'object_metadata_%' AND Table_type = 'BASE TABLE'"
+            );
+        foreach ($metaDataTables as $table) {
+            $tableName = current($table);
+            $metaDataTable = $schema->getTable($tableName);
+
+            if (!$metaDataTable->hasColumn(self::AUTO_ID)) {
+                $metaDataTable->addColumn(self::AUTO_ID, 'integer', [
+                    'autoincrement' => true,
+                ]);
+
+                $fkName = AbstractDao::getForeignKeyName($tableName, self::ID_COLUMN);
+                $this->addSql('ALTER TABLE `' . $tableName . '` DROP FOREIGN KEY IF EXISTS `' . $fkName . '`');
+                $metaDataTable->dropPrimaryKey();
+                $metaDataTable->setPrimaryKey([self::AUTO_ID]);
+                $metaDataTable->addUniqueIndex(self::PK_COLUMNS, self::UNIQUE_INDEX_NAME);
+
+                $metaDataTable->addForeignKeyConstraint(
+                    'objects',
+                    [self::ID_COLUMN],
+                    [self::ID_COLUMN],
+                    ["onDelete" => "CASCADE"],
+                    $fkName
+                );
+            }
+        }
+
+        $this->addSql('SET foreign_key_checks = 1');
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->addSql('SET foreign_key_checks = 0');
+
+        $db = Db::get();
+
+        $metaDataTables = $db->fetchAllAssociative(
+            "SHOW FULL TABLES
+                       WHERE `Tables_in_{$db->getDatabase()}` LIKE 'object_metadata_%' AND Table_type = 'BASE TABLE'"
+        );
+        foreach ($metaDataTables as $table) {
+            $metaDataTable = $schema->getTable(current($table));
+
+            if ($metaDataTable->hasColumn(self::AUTO_ID)) {
+                $metaDataTable->dropPrimaryKey();
```

Comment:
facing an issue when trying to `migrate --down`
To reproduce, migrate up, change an object to have a `adv.many to many` field and save it, trying to run migration down `bin/console d:m:e "Pimcore\Bundle\CoreBundle\Migrations\Version20230616085142" --down`
Only crashes on the freshly created table
![image](https://github.com/pimcore/pimcore/assets/6014195/1e230ecf-62ed-4a2b-b34e-eaa90261b606)


probably needs
```suggestion
                $this->addSql('ALTER TABLE `' . $tableName . '` DROP FOREIGN KEY IF EXISTS `' . $fkName . '`');                
                $metaDataTable->dropPrimaryKey();
```
and readd it by addForeignKeyConstraint

Reply:
Found a discrepancy checking with PhpStorm SQL Generator: on the left, it's a migrated table and on the right is a created by `DAO::createOrUpdateTable()`

![image](https://github.com/pimcore/pimcore/assets/6014195/ed86c9d4-5254-4489-913e-02ffcb28b450)
beside the `id` column order should be put at first position in migration, there are `fk_object_metadata_NE__o_id` and `IDX_208F4545DB01246B`